pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -271,13 +271,16 @@ func (service *AlertRuleService) ReplaceRuleGroup(ctx context.Context, orgID int
 
 		updates := make([]models.UpdateRule, 0, len(delta.Update))
 		for _, update := range delta.Update {
-			// check that provenance is not changed in a invalid way
+			// check that provenance is not changed in an invalid way
 			storedProvenance, err := service.provenanceStore.GetProvenance(ctx, update.New, orgID)
 			if err != nil {
 				return err
 			}
-			if storedProvenance != provenance && storedProvenance != models.ProvenanceNone {
+			err = canUpdateWithProvenance(storedProvenance, provenance, func() error {
```

Comment:
I had to allow downgrading from provenance `API` to `None` in `ReplaceRuleGroup` as this method is a `replace`.

I understand this as if we removed all rules and inserted them again. So, this had to be done to allow insertions, updates, and deletes inside the method.